dotCMS/src/test/java/com/dotcms/filters/NormalizationFilterTest.java

```diff
@@ -0,0 +1,115 @@
+        //Capturing the request when is passed down to the chain
+        doNothing().when(chain).doFilter(capturedRequest.capture(), any(HttpServletResponse.class));
+    }
+
+    /**
+     * Test to verify the {@link NormalizationFilter} is applying properly the normalization on
+     * invalids URIs
+     */
+    @Test
+    public void test_uri_normalization_invalid_URI() throws IOException, ServletException {
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        String originalURI = "/test/../folder/important/secret_file.dat";
+        String expectedNormalizedURI = "/folder/important/secret_file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        originalURI = "/test/../folder/folder1/forward_jsp.jsp?FORWARD_URL=http://google.com";
+        expectedNormalizedURI = "/folder/folder1/forward_jsp.jsp?FORWARD_URL=http://google.com";
+        compare(originalURI, expectedNormalizedURI);
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        originalURI = "/test/../folder/important/../secret_file.dat";
+        expectedNormalizedURI = "/folder/secret_file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // Each "." segment is simply removed
+        originalURI = "./folder/folder1/file.dat";
+        expectedNormalizedURI = "folder/folder1/file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // Each "." segment is simply removed
+        originalURI = "./folder/./folder1/file.dat";
+        expectedNormalizedURI = "folder/folder1/file.dat";
+        compare(originalURI, expectedNormalizedURI);
+
+        // Each "." segment is simply removed
+        originalURI = "/folder/./folder1/file.dat";
+        expectedNormalizedURI = "/folder/folder1/file.dat";
+        compare(originalURI, expectedNormalizedURI);
+    }
+
+    /**
+     * Test to verify the {@link NormalizationFilter} is applying properly the normalization on
+     * valid URIs
+     */
+    @Test
+    public void test_uri_normalization_valid_URI() throws IOException, ServletException {
+
+        String originalURI = "/folder/important/secret_file.dat";
+        compare(originalURI, originalURI);
+
+        originalURI = "/folder/folder1/forward_jsp.jsp?FORWARD_URL=http://google.com";
+        compare(originalURI, originalURI);
+
+        originalURI = "folder/folder1/file.dat";
+        compare(originalURI, originalURI);
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        originalURI = "../folder/folder1/file.dat";
+        compare(originalURI, originalURI);
+    }
+
+    private void compare(final String originalURI, final String expectedNormalizedURI)
+            throws IOException, ServletException {
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'request' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4798850)